src/metabase/lib/column_group.cljc

```diff
@@ -94,7 +96,18 @@
 (defmethod display-info-for-group-method :group-type/join.implicit
   [query stage-number {:keys [fk-field-id], :as _column-group}]
   (merge
-   (when-let [field (lib.metadata/field query fk-field-id)]
+   (when-let [field (if (string? fk-field-id)
+                      ;; TODO: This is probably working around a QP bug. See #37067 - the `data.cols` from the API has
+                      ;; fk_field_id: "customer_id" but I think that's wrong and it should not have dropped the real
+                      ;; field ID somewhere in the QP.
+                      (some->> (lib.util/query-stage query stage-number)
+                               (lib.metadata.calculation/visible-columns query stage-number)
+                               (lib.equality/find-matching-column
+                                 query stage-number
+                                 (lib.options/ensure-uuid [:field {:base-type :type/Integer} fk-field-id]))
```

Comment:
Are all FKs integers?

Reply:
Definitely not, but I had to say something. I suppose it should be `:type/*` instead.